tests/zfs-tests/tests/functional/rsend/rsend.kshlib

```diff
@@ -551,3 +551,120 @@ function test_fs_setup
 	fi
 	log_must zfs create -o compress=lz4 $sendpool/stream
 }
+
+#
+# Check to see if the specified features are set in a send stream.
+# The values for these features are found in include/sys/zfs_ioctl.h
+#
+# $1 The stream file
+# $2-$n The flags expected in the stream
+#
+function stream_has_features
+{
+	typeset file=$1
+	shift
+
+	[[ -f $file ]] || log_fail "Couldn't find file: $file"
+	typeset flags=$(cat $file | zstreamdump | awk '/features =/ {print $3}')
```

Comment:
I assume you disabled "send-cpL_varied_recsize" because you're getting "arithmetic syntax error" from ksh.

This seems to happen when we have multiple BEGIN records and the `flags` variable becomes a multiline string:

```
root@debian-8-zfs:~# zfs create $POOLNAME/fs
root@debian-8-zfs:~# zfs create $POOLNAME/fs/sub1
root@debian-8-zfs:~# zfs create $POOLNAME/fs/sub2
root@debian-8-zfs:~# zfs snap -r $POOLNAME/fs@snap1
root@debian-8-zfs:~# zfs send -cpLRDe $POOLNAME/fs@snap1 > $TMPDIR/zstream.dat
root@debian-8-zfs:~# #
root@debian-8-zfs:~# cat $TMPDIR/zstream.dat | zstreamdump | awk '/features =/ {print $3}'
7
430007
430007
430007
root@debian-8-zfs:~# ksh
# flags='7
> 430007
> 430007
> 430007'
# 
# printf "%x" $((0x$flags & 0x430007))
ksh: 0x7
430007
430007
430007 & 0x430007: arithmetic syntax error
# 
```

`mdb` seems to consider only the ending line:

```
[root@52-54-00-d3-7a-f3 ~]# echo -e "0\n0\n4 & 4 = X" | mdb
                4               
[root@52-54-00-d3-7a-f3 ~]# echo -e "0\n4\n0 & 4 = X" | mdb
                0               
[root@52-54-00-d3-7a-f3 ~]# echo -e "4\n0\n0 & 4 = X" | mdb
                0               
[root@52-54-00-d3-7a-f3 ~]# 
```

We should be able to "fix" the issue like this:

```sh
typeset flags=$(cat $file | zstreamdump | awk '/features =/ {features = $3} END {print features}')
```